### PR TITLE
Refactor AttributeManager

### PR DESCRIPTION
**Pull request by kchabSS**
Refactor AttributeManager along with all connected parts to:
* rearrange parameters to move all commonly used parameters to the left.
* use DynTypedNode
* merge all to a single map with a shared key

### CODE REVIEW

**Review by IuriiKobein — COMMENTED** [review]
Nice work +980 −1,910  loc :)
Without deep dive into detail I have few requests:
 - please move rename files, dirs and class from 'attribute manager' to 'handler manger'
 - move and rename ast_preseccor to be under the 'sema' dir
 - introduce static methods to handler manager to hide access to the singleton

**Comment by kchabSS** [comment]
> Nice work +980 −1,910 loc :) Without deep dive into detail I have few requests:
> 
> * please move rename files, dirs and class from 'attribute manager' to 'handler manger'
OK. Sounds nice. I use `HandlerMap` as storage and call router.
> * move and rename ast_preseccor to be under the 'sema' dir
OK. Will do.
> * introduce static methods to handler manager to hide access to the singleton
OK. No more `Singleton` instance? Can also move `SessionStage` to constructor parameter, to not carry it all the time around the calls?

**Review by IuriiKobein — COMMENTED** [review]
It looks like not all new files are not committed:
 handler_map.cpp

**Comment by kchabSS** [comment]
> It looks like not all new files are not committed: handler_map.cpp

Ups. Fixed.

File: lib/core/transpiler_session/attributed_type_map.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #pragma once
 
-#include <clang/AST/ASTTypeTraits.h>
-#include <llvm/ADT/DenseSet.h>
+#include "../../../../../../../usr/lib/llvm-17/include/clang/AST/ASTTypeTraits.h"
```
Comment (IuriiKobein) [inline lib/core/transpiler_session/attributed_type_map.h:3]: It looks not good, probably some debug footprints

**Review by IuriiKobein — COMMENTED** [review]
Looks nice.

File: lib/core/transpiler_session/attributed_type_map.h (hunk 1 above)
Comment (kchabSS) [reply lib/core/transpiler_session/attributed_type_map.h:3]: Nice find. I missed it probably during refactoring.

Comment (kchabSS) [reply lib/core/transpiler_session/attributed_type_map.h]: Fixed.

**Review by IuriiKobein — APPROVED** [review]

**Pull request merged**